src/PublisherInterface.php

```diff
@@ -20,5 +20,5 @@
  */
 interface PublisherInterface
 {
-    public function __invoke(Update $update): string;
+    public function publish(Update $update): string;
```

Comment:
this is the only BC-breaking change.

how do you suggest we avoid this @dunglas ?

Reply:
Usually, we handle such cases by making the new method "virtual" until the next major, which means we only add an `@method` phpdoc annotation on the interface. 
IIRC, IDEs will complain when a userland implementation does not implement it yet. More importantly, the Symfony `DebugClassLoader` will automatically trigger a deprecation notice telling that not implementing this method is deprecated. 
Then add a comment to the `__invoke()` description saying that it will be removed in favor of `publish()` in 1.0.
I think such a BC layer is more than acceptable for a library that has no `1.x` release yet (we could not care about BC at all here).

Reply:
Aha, thanks for the idea 👍 

but i think according to semver, we are allowed to break BC between 0.4 and 0.5, so i think we can ignore this, as this would only effect people implementing their own publishers, which is highly unlikely ( until someone comes saying we broke their app 😛 ), as for the other things that changes ( e.g providers, publisher constructor .. etc ), there's a BC layer which allow people to use both 0.4 and 0.5 safely.

Reply:
Re-reading the code, everything is marked as experimental yes, so we might not need any deprecation at all. 
I'll let @dunglas advise here

Reply:
According to semver we can, but this one is a big BC break that will impact many existing projects (especially in the API Platform community). I would prefer to provide an upgrade path... or to keep it as is (I prefer `publish()` but I quite like using `invoke()` too).

I'm in favor of doing what @chalasr explained.

Reply:
done 👍